aten/src/ATen/native/hammerblade/Eye.cpp

```diff
@@ -0,0 +1,25 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/hammerblade/HammerBladeContext.h>
+#include <ATen/native/hammerblade/Offload.h>
+
+namespace at { 
+namespace native {
+
+  Tensor& eye_out_hb(Tensor& output, long n, long m) {
+    std::vector<eva_t> device_args;
+    std::vector<eva_t> device_ptrs;
+    if(m == -1) {
+      m = n;
+    }
+    output.resize_({n,m});
+    device_args.push_back(create_device_tensor(output, device_ptrs));
```

Comment:
I think you don't have to call these low level APIs yourself? `hb_offload_kernel` will take care of them for you.
You can write
```C
hb_offload_kernel(output, n, m, "tensorlib_eye");
```

Reply:
Fixed!